### PR TITLE
Fix parse api

### PR DESCRIPTION
**Pull request by film42**
This is a breaking change.

1. Instead of returning a hash, this will now return `{:ok, _}` to match the error/ reason pattern.
2. This changes `parse` to return `%{signon: %{}, accounts: [%{}, %{}, %{}, ...]}`.

Let me know what you think :).

cc @jjcarstens

### CODE REVIEW

File: lib/ofex.ex [hunk 1]
```diff
@@ -47,14 +47,21 @@ defmodule Ofex do
...
+        |> Enum.reduce(%{signon: %{}, accounts: []}, &reduce_items_list/2)
+
+        {:ok, result}
       {:error, message} -> {:error, %InvalidData{message: message, data: data}}
     end
   end
 
+  defp reduce_items_list(%{signon: signon}, %{accounts: accounts}), do: %{signon: signon, accounts: accounts}
+  defp reduce_items_list(%{account: account}, %{accounts: accounts}=acc), do: Map.put(acc, :accounts, [account | accounts])
+  defp reduce_items_list(_, acc), do: acc
```
Comment (jjcarstens) [inline lib/ofex.ex]: Not totally in love with the function name, but not sure I have a better idea. Maybe something like `accumulate_parsed_items`?

File: lib/ofex.ex [hunk 2]
```diff
@@ -47,14 +47,21 @@ defmodule Ofex do
   def parse(data) do
     case validate_ofx_data(data) do
       {:ok, parsed_ofx} ->
-        parsed_ofx
+        result = parsed_ofx
         |> xpath(~x"//OFX/*[contains(name(),'MSGSRS')]"l)
         |> Enum.map(&parse_message_set(xpath(&1, ~x"name()"s), &1))
-        |> Map.new
+        |> List.flatten
+        |> Enum.reduce(%{signon: %{}, accounts: []}, &reduce_items_list/2)
```
Comment (jjcarstens) [inline lib/ofex.ex]: Since we're assigning this now, can we space all the `|>` piped commands under `parsed_ofx`

**Review by jjcarstens — COMMENTED** [review]
👍 For `{:ok, parsed}` return. Much better pattern to use with a `{:error, error}` return to keep consistent.

I should probably get this hooked up to some CI..

**Comment by film42** [comment]
Fixed!

**Review by jjcarstens — APPROVED** [review]
💥 :shipit:

**Pull request merged**